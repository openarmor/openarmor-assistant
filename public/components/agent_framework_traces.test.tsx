/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { render, screen } from '@testing-library/react';
import { AgentFrameworkTraces } from './agent_framework_traces';
import * as GetTraces from '../hooks/use_fetch_agentframework_traces';

describe('<AgentFrameworkTraces/> spec', () => {
  it('renders the component', async () => {
    const traces = [
      {
        interactionId: 'test_interactionId',
        input: 'input',
        output: 'output',
        createTime: '',
        origin: '',
        traceNumber: 1,
      },
      {
        interactionId: 'test_interactionId',
        input: 'input',
        output: 'output',
        createTime: '',
        origin: 'CatIndexTool',
        traceNumber: 2,
      },
      {
        interactionId: 'test_interactionId',
        input: 'input',
        output: '',
        createTime: '',
        origin: '',
        traceNumber: 3,
      },
      {
        interactionId: 'test_interactionId',
        input: '',
        output: 'output',
        createTime: '',
        origin: '',
        traceNumber: 4,
      },
    ];
    const mockedGetTracesResult = {
      loading: false,
      data: traces,
    };

    jest.spyOn(GetTraces, 'useFetchAgentFrameworkTraces').mockReturnValue(mockedGetTracesResult);

    render(<AgentFrameworkTraces interactionId="test-interaction-id" />);
    expect(GetTraces.useFetchAgentFrameworkTraces).toBeCalledTimes(1);
    expect(document.body.children).toMatchSnapshot();
  });

  it('no traces available', async () => {
    jest.spyOn(GetTraces, 'useFetchAgentFrameworkTraces').mockReturnValue({
      loading: false,
      data: [],
    });
    render(<AgentFrameworkTraces interactionId="test-interaction-id" />);
    expect(screen.queryByText('Data not available.')).toBeInTheDocument();
  });

  it('show loading', async () => {
    jest.spyOn(GetTraces, 'useFetchAgentFrameworkTraces').mockReturnValue({
      loading: true,
      data: [],
    });
    render(<AgentFrameworkTraces interactionId="test-interaction-id" />);
    expect(screen.queryByText('Loading...')).toBeInTheDocument();
  });

  it('show error', async () => {
    jest.spyOn(GetTraces, 'useFetchAgentFrameworkTraces').mockReturnValue({
      loading: false,
      data: [],
      error: new Error('test'),
    });
    render(<AgentFrameworkTraces interactionId="test-interaction-id" />);
    expect(screen.queryByText('Error loading details')).toBeInTheDocument();
  });
});
