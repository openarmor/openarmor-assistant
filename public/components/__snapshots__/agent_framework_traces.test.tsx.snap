// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AgentFrameworkTraces/> spec renders the component 1`] = `
HTMLCollection [
  <div>
    <div
      class="euiMarkdownFormat"
    >
      <div>
        <h1
          id="how-was-this-generated"
        >
          How was this generated
        </h1>
        

        <h4
          id="question"
        >
          Question
        </h4>
        

        <h4
          id="result"
        >
          Result
        </h4>
        

        <p>
          output
        </p>
      </div>
    </div>
    <div
      class="euiSpacer euiSpacer--l"
    />
    <div
      class="euiText euiText--medium"
    >
      <h3>
        Response
      </h3>
    </div>
    <div>
      <div
        class="euiSpacer euiSpacer--s"
      />
      <div
        class="euiAccordion"
      >
        <div
          class="euiAccordion__triggerWrapper"
        >
          <button
            aria-controls="Step 1 - CatIndexTool"
            aria-expanded="false"
            class="euiAccordion__button"
            id="random_html_id"
            type="button"
          >
            <span
              class="euiAccordion__iconWrapper"
            >
              <svg
                aria-hidden="true"
                class="euiIcon euiIcon--medium euiIcon-isLoading euiAccordion__icon"
                focusable="false"
                height="16"
                role="img"
                viewBox="0 0 16 16"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M5.277 10.088c.02.014.04.03.057.047.582.55 1.134.812 1.666.812.586 0 1.84-.293 3.713-.88L9 6.212V2H7v4.212l-1.723 3.876Zm-.438.987L3.539 14h8.922l-1.32-2.969C9.096 11.677 7.733 12 7 12c-.74 0-1.463-.315-2.161-.925ZM6 2H5V1h6v1h-1v4l3.375 7.594A1 1 0 0 1 12.461 15H3.54a1 1 0 0 1-.914-1.406L6 6V2Z"
                />
              </svg>
            </span>
            <span
              class="euiIEFlexWrapFix"
            >
              Step 1 - CatIndexTool
            </span>
          </button>
        </div>
        <div
          aria-labelledby="random_html_id"
          class="euiAccordion__childWrapper"
          id="Step 1 - CatIndexTool"
          role="region"
          tabindex="-1"
        >
          <div>
            <div
              class=""
            >
              <div
                class="euiCodeBlock euiCodeBlock--fontMedium euiCodeBlock--paddingSmall prismjs language-none"
              >
                <pre
                  class="euiCodeBlock__pre euiCodeBlock__pre--whiteSpacePreWrap"
                  tabindex="-1"
                >
                  <code
                    class="euiCodeBlock__code"
                  >
                    Input: 
                    input
                  </code>
                </pre>
              </div>
              <div
                class="euiCodeBlock euiCodeBlock--fontMedium euiCodeBlock--paddingSmall prismjs language-none"
              >
                <pre
                  class="euiCodeBlock__pre euiCodeBlock__pre--whiteSpacePreWrap"
                  tabindex="-1"
                >
                  <code
                    class="euiCodeBlock__code"
                  >
                    Output: 
                    output
                  </code>
                </pre>
              </div>
            </div>
          </div>
        </div>
      </div>
      <hr
        class="euiHorizontalRule euiHorizontalRule--full euiHorizontalRule--marginXSmall"
      />
    </div>
  </div>,
]
`;
